tools/interop_matrix/client_matrix.py

```diff
@@ -81,7 +81,7 @@ def get_github_repo(lang):
         'v1.6.6',
     ],
     'ruby': [
-        # Ruby v1.0.x doesn't have the fix #8914, therefore not supported.
+        'v1.0.1',
```

Comment:
IMHO sth like this would a nice way to express the need for patching - it is readable and can be used for any language and release.
```
 'ruby': [
      ('v1.0.1', {'patches': ['some_patch_under_patches_directory.patch']}),
      ('v1.1.4', {}),
      'v1.2.5'
     ....
```
or maybe 
```
 'ruby': {
      {'version': 'v1.0.1', 'patches': ['some_patch_under_patches_directory.patch']},
     ....
```

Reply:
I went with a slight tweak to the latter suggestion.

It now looks like:

```
'ruby': {
      {'version': 'v1.0.1', 'patches': [<files_to_patch>,....]},
```

and then each patch file is named `git_repo.patch`, under the `patches/<lang>_<release>` subdirectory.

I liked this approach because it gives the flexibility for a patch to modify the whole git repo at a certain tag if it needs to and also add new files to git.